server/chat-commands/admin.ts

```diff
@@ -735,18 +737,18 @@ export const commands: ChatCommands = {
 
 	async updateserver(target, room, user, connection) {
 		if (!this.canUseConsole()) return false;
-
-		if (Monitor.updateServerLock) {
+		const isPrivate = toID(target) === 'private';
+		if (Monitor.updateServerLock === (isPrivate ? 'private' : 'public')) {
```

Comment:
I think having separate private/public locks is overkill, but even if it's not, this is wrong since updating private would override the public lock and vice versa.

Reply:
That's a... large mistake on my part. Fixed (and yeah thinking about how often /updateserver is used, we probably don't need to support updating private and public code concurrently, although it wouldn't be hard).